control-plane/csi-driver/src/bin/node/main_.rs

```diff
@@ -345,37 +382,59 @@ impl CsiServer {
 }
 
 /// Gets the nvme ana multipath parameter and sets on the node as a label.
-async fn check_ana_and_label_node(node_name: &str) -> anyhow::Result<()> {
-    let patch_value = utils::check_nvme_core_ana().unwrap_or_default().to_string();
-    if let Ok(client) = Client::try_default().await {
-        let nodes: Api<K8sNode> = Api::all(client);
-        let node_patch = json!({
-            "apiVersion": "v1",
-            "kind": "Node",
-            "metadata": {
-                "labels": {
-                    utils::CSI_NODE_NVME_ANA: patch_value,
-                },
+async fn check_ana_and_label_node(
+    client: &kube::client::Client,
+    node_name: &str,
+    nvme_enabled: String,
+) -> anyhow::Result<()> {
+    let node_patch = json!({
+        "apiVersion": "v1",
+        "kind": "Node",
+        "metadata": {
+            "labels": {
+                utils::CSI_NODE_NVME_ANA: nvme_enabled,
             },
-        });
-        match nodes
-            .patch(
-                node_name,
-                &PatchParams::apply("node_label_patch").force(),
-                &Patch::Apply(&node_patch),
-            )
-            .await
-        {
-            Ok(_) => trace!("Patched node: {} with patch: {}", node_name, node_patch),
-            Err(error) => anyhow::bail!(
-                "Failed to patch node: {} with patch: {}. {}",
-                node_name,
-                node_patch,
-                error
-            ),
-        }
+        },
+    });
+    patch_k8s_node(client, node_name, &node_patch).await?;
+    Ok(())
+}
+
+/// Validate that the instance endpoint and grpc endpoint are valid, and returns the instance
+/// endpoint.
+fn validate_endpoints(
+    instance_endpoint: &str,
+    grpc_endpoint: &str,
+) -> anyhow::Result<(SocketAddr, SocketAddr)> {
+    // Append the port to the grpc endpoint if it is not specified.
+    let grpc_endpoint = if grpc_endpoint.contains(':') {
+        grpc_endpoint.to_string()
     } else {
-        anyhow::bail!("Failed to get an instance of k8s client")
+        format!("{grpc_endpoint}:{GRPC_PORT}")
+    };
+
+    // Append the port to the instance endpoint with the grpc endpoint's port if it is not
+    // specified.
+    let instance_endpoint = if instance_endpoint.contains(':') {
+        instance_endpoint.to_string()
+    } else {
+        format!(
+            "{instance_endpoint}:{}",
+            grpc_endpoint
+                .split(':')
+                .last()
+                .ok_or(anyhow!("gRPC endpoint must have a port"))?
+        )
+    };
+
+    let instance_url = SocketAddr::from_str(&instance_endpoint)?;
+    let grpc_endpoint_url = SocketAddr::from_str(&grpc_endpoint)?;
+
+    if instance_url.port() != grpc_endpoint_url.port() {
+        return Err(anyhow!(
+            "instance endpoint and gRPC endpoint must have the same port"
```

Comment:
Seems a bit pointless to do this.
Also what if the ip is different?
Why can't we just reuse grpc_endpoint?

Reply:
No we can't use the same endpoint, grpc endpoint can be 0.0.0.0:<port>, but the instance endpoint always needs to the ip of the instance. That's why grpc endpoint is a non required field.

Reply:
Why don't we change grpc endpoint to not be 0.0.0.0 and be MY_POD_IP?

Reply:
But we still need a default value right?

Reply:
Or we can make it required with the MY_POD_IP maybe

Reply:
Done.